prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -229,6 +238,95 @@ class ReportFunction(
         }
     }
 
+    class SftpSubmissionException(override val message: String) : RuntimeException(message)
+
+    @FunctionName("submitSFTP")
+    fun submitViaSftp(
+        @BlobTrigger(
+            name = "report",
+            dataType = "string",
+            path = "sftp-submissions/{name}.{extension}",
+            connection = "SftpStorage"
+        ) content: String,
+        @BindingName("name") name: String,
+        @BindingName("extension") extension: String,
+    ) {
+        val format = try {
+            MimeFormat.valueOfFromExt(extension)
+        } catch (ex: IllegalArgumentException) {
+            throw SftpSubmissionException("$extension is not valid.")
+        }
+
+        val filename = "$name.$extension"
+
+        val senderId = getSenderIdFromFilePath(filename)
+        val sender = workflowEngine.settings.findSender(senderId)
+            ?: throw SftpSubmissionException("No sender found for $senderId, parsed from $filename")
+
+        if (sender.customerStatus == CustomerStatus.INACTIVE) {
+            logger.info("Sender is disabled, not processing the report")
+            // TODO https://github.com/CDCgov/prime-reportstream/issues/16260
+            return
+        }
+
+        val payloadName = extractPayloadNameFromFilePath(filename)
+        actionHistory.trackActionSenderInfo(sender.fullName, payloadName)
+
+        try {
+            val receiver = SubmissionReceiver.getSubmissionReceiver(sender, workflowEngine, actionHistory)
+            val rawBody = content.toByteArray()
+            val report = receiver.validateAndMoveToProcessing(
+                sender,
+                content,
+                emptyMap(),
+                Options.None,
+                emptyList(),
+                isAsync = true,
+                allowDuplicates = true,
+                rawBody = rawBody,
+                payloadName = payloadName
+            )
+
+            reportEventService.sendReportEvent(
+                eventName = ReportStreamEventName.REPORT_RECEIVED,
+                childReport = report,
+                pipelineStepName = TaskAction.receive
+            ) {
+                params(
+                    listOfNotNull(
+                        ReportStreamEventProperties.SENDER_NAME to sender.fullName,
+                        ReportStreamEventProperties.INGESTION_TYPE to IngestionMethod.SFTP,
+                        ReportStreamEventProperties.ITEM_FORMAT to format
+                    ).toMap()
+                )
+            }
+        } catch (e: ActionError) {
+            actionHistory.trackLogs(e.details)
+        } catch (e: IllegalArgumentException) {
+            actionHistory.trackLogs(
```

Comment:
We don't want poison queue functionality for the cases here?

Reply:
I could see it going either way.  In the REST scenario we return a `BAD_REQUEST` response synchronously.  I'm happy to just rethrow